src/journal/journal-file.c

```diff
@@ -194,6 +195,22 @@ JournalFile* journal_file_close(JournalFile *f) {
         return NULL;
 }
 
+static void * journal_file_close_async_thread(JournalFile *f) {
+        pthread_detach(pthread_self());
+
+        journal_file_close(f);
+
+        pthread_exit(NULL);
+}
+
+static void journal_file_close_async(JournalFile *f) {
+        pthread_t t;
+
+        assert(f);
+
+        pthread_create(&t, NULL, (void *(*)(void *))journal_file_close_async_thread, f);
```

Comment:
Please make `journal_file_close_async_thread()` a `void *(*start_routine) (void *)` prototype and do the cast inside that function. Also, `pthread_create()` has a return value which needs to be passed up the chain from this function.